detection_rules/rule.py

```diff
@@ -585,17 +595,47 @@ def lock_info(self, bump=True) -> dict:
     @property
     def is_dirty(self) -> Optional[bool]:
         """Determine if the rule has changed since its version was locked."""
-        min_stack = self.metadata.get('min_stack_version') or str(get_min_supported_stack_version(drop_patch=True))
-        existing_sha256 = self.version_lock.get_locked_hash(self.id, min_stack)
+        existing_sha256 = self.version_lock.get_locked_hash(self.id, self.get_stack_version())
```

Comment:
If I understand correctly, these 17 rules had changes to them since we manually forked and changed versions to 100. As a result, 8.3 was being used as the min_stack and thus the comparison of the 7.16 and 8.3 hash differed so it suspected it of being dirty. Rather if it were to compare it to itself, as `self.get_stack_version()` is allowing, there is no diff and the forked version is not bumped?

Reply:
Most were non-7.16. There were a few 7.16 and the bug was a diff in the version between the rule and the lock (7.16 vs 7.16.0)